crates/task-impls/src/request.rs

```diff
@@ -160,237 +159,106 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TaskState for NetworkRequest
-/// a request.  If at any point the requested info is seen in the data stores or
-/// the view has moved beyond the view we are requesting, the task will completed.
-struct DelayedRequester<TYPES: NodeType, I: NodeImplementation<TYPES>> {
-    /// The underlying network to send requests on
-    pub network: Arc<I::Network>,
-    /// Shared state to check if the data go populated
-    state: OuterConsensus<TYPES>,
-    /// our public key
-    public_key: TYPES::SignatureKey,
-    /// Channel to send the event when we receive a response
-    sender: Sender<Arc<HotShotEvent<TYPES>>>,
-    /// Duration to delay sending the first request
-    delay: Duration,
-    /// The peers we will request in a random order
-    recipients: Vec<TYPES::SignatureKey>,
-    /// A flag indicating that `HotShotEvent::Shutdown` has been received
-    shutdown_flag: Arc<AtomicBool>,
-    /// The node's id
-    id: u64,
-}
-
-/// Wrapper for the info in a VID request
-struct VidRequest<TYPES: NodeType>(TYPES::Time, TYPES::SignatureKey);
-
-impl<TYPES: NodeType, I: NodeImplementation<TYPES>> DelayedRequester<TYPES, I> {
-    /// Wait the delay, then try to complete the request.  Iterates over peers
-    /// until the request is completed, or the data is no longer needed.
-    async fn run(self, request: RequestKind<TYPES>, signature: Signature<TYPES>) {
-        match request {
-            RequestKind::Vid(view, key) => {
-                // Do the delay only if primary is up and then start sending
-                if !self.network.is_primary_down() {
-                    async_sleep(self.delay).await;
-                }
-                self.do_vid(VidRequest(view, key), signature).await;
+        let network = Arc::clone(&self.network);
+        let delay = self.delay;
+        async_spawn(async move {
+            // Do the delay only if primary is up and then start sending
+            if !network.is_primary_down() {
+                async_sleep(delay).await;
             }
-            RequestKind::Proposal(..) | RequestKind::DaProposal(..) => {}
-        }
-    }
-    /// Handle sending a VID Share request, runs the loop until the data exists
-    async fn do_vid(&self, req: VidRequest<TYPES>, signature: Signature<TYPES>) {
-        let message = make_vid(&req, signature);
-        let mut recipients_it = self.recipients.iter().cycle();
-
-        let serialized_msg = match bincode::serialize(&message) {
-            Ok(serialized_msg) => serialized_msg,
-            Err(e) => {
-                tracing::error!(
-                    "Failed to serialize outgoing message: this should never happen. Error: {e}"
-                );
+            broadcast_event(
+                HotShotEvent::VidRequestSend(request, signature).into(),
+                &sender,
+            )
+            .await;
 
-                return;
-            }
-        };
+            // start waiting
+            for _ in 0..4 {
```

Comment:
Why do we have 0..4 here? We could instead just wrap this in a `loop` statement or `while` like we do with the proposal request, then wrap *that* loop in an `async_timeout` that way it'll just be reaped if it's not done in time.

Reply:
yea this was just testing, thats a good idea!